src/MiniProfiler.Providers.StackExchange.Redis/RedisStorage.cs

```diff
@@ -0,0 +1,263 @@
+        /// <param name="user">The user to set this profiler ID as viewed for.</param>
+        /// <param name="id">The profiler ID to set viewed.</param>
+        public void SetViewed(string user, Guid id)
+        {
+            RedisKey key = ProfilerResultUnviewedSetKeyPrefix.Append(user);
+            RedisValue value = id.ToString();
+            _database.SetRemove(key, value);
+        }
+
+        /// <summary>
+        /// Returns a list of <see cref="MiniProfiler.Id"/>s that haven't been seen by <paramref name="user"/>.
+        /// </summary>
+        /// <param name="user">User identified by the current <c>MiniProfiler.Settings.UserProvider</c></param>
+        public List<Guid> GetUnviewedIds(string user)
+        {
+            RedisKey key = ProfilerResultUnviewedSetKeyPrefix.Append(user);
+            var ids = _database.SetMembers(key);
+            return ids.Select(x => Guid.Parse(x)).ToList();
+        }
+
+        /// <summary>
+        /// Asynchronously list the latest profiling results.
+        /// </summary>
+        /// <param name="maxResults">The maximum number of results to return.</param>
+        /// <param name="start">(Optional) The start of the date range to fetch.</param>
+        /// <param name="finish">(Optional) The end of the date range to fetch.</param>
+        /// <param name="orderBy">(Optional) The order to fetch profiler IDs in.</param>
+        public async Task<IEnumerable<Guid>> ListAsync(
+            int maxResults,
+            DateTime? start = null,
+            DateTime? finish = null,
+            ListResultsOrder orderBy = ListResultsOrder.Descending)
+        {
+            // TODO: start/finish not yet supported - will simply return all results
+
+            IEnumerable<RedisValue> ids;
+            switch (orderBy)
+            {
+                case ListResultsOrder.Ascending:
+                    ids = (await _database.ListRangeAsync(ProfilerResultListKey, maxResults + 1, -1).ConfigureAwait(false)).Reverse();
+                    break;
+                case ListResultsOrder.Descending:
+                default:
+                    ids = await _database.ListRangeAsync(ProfilerResultListKey, 0, maxResults - 1).ConfigureAwait(false);
+                    break;
+            }
+            return ids.Select(x => Guid.Parse(x));
+        }
+
+        /// <summary>
+        /// Asynchronously stores <paramref name="profiler"/> under its <see cref="MiniProfiler.Id"/>.
+        /// </summary>
+        /// <param name="profiler">The <see cref="MiniProfiler"/> to save.</param>
+        /// <remarks>
+        /// Should also ensure the profiler is stored as being un-viewed by its profiling <see cref="MiniProfiler.User"/>.
+        /// </remarks>
+        public async Task SaveAsync(MiniProfiler profiler)
+        {
+            var id = profiler.Id.ToString();
+            RedisKey key = ProfilerResultKeyPrefix.Append(id);
+            RedisValue value = profiler.ToJson();
```

Comment:
See below comment on Protobuf serialization (@mgravell is this build into SE.Redis anywhere at this point?)